lib/mayu/resources/mermaid_exporter.rb

```diff
@@ -0,0 +1,203 @@
+        out.puts "  end"
+
+        print_route_edges(out, tree.dig("", "app", "pages") || {})
+
+        print_subgraphs(out, tree)
+
+        entries.each do |entry|
+          @graph
+            .direct_dependencies_of(entry)
+            .each { |dep| out.puts "  #{encode(entry)}-->#{encode(dep)}" }
+        rescue StandardError
+          next
+        end
+
+        entries.each do |entry|
+          unless @graph.get_resource(entry)&.exists?
+            out.puts "  class #{encode(entry)} NonExistant"
+          end
+
+          filetype_class(entry)&.tap do
+            out.puts "  class #{encode(entry)} #{_1}"
+          end
+        end
+
+        out.puts <<~EOF.gsub(/^/, "  ")
+        style routes stroke:#09c,stroke-width:5,fill:#f0f;
+        classDef cluster fill:#0003;
+        classDef Ruby fill:#600,stroke:#900,stroke-width:3px;
+        classDef Image fill:#069,stroke:#09c,stroke-width:3px;
+        classDef CSS fill:#063,stroke:#096,stroke-width:3px;
+        classDef NonExistant opacity:50%,stroke-dasharray:5px;
+        linkStyle default fill:transparent,opacity:50%;
+      EOF
+
+        out
+      end
+
+      private
+
+      sig { params(path: String).returns(T.nilable(String)) }
+      def filetype_class(path)
+        case File.extname(path)
+        when ".rb"
+          "Ruby"
+        when ".css"
+          "CSS"
+        when ".png"
+          "Image"
+        end
+      end
+
+      sig { params(str: String).returns(String) }
+      def encode(str)
+        str.gsub("/", "__").gsub("[", "__").gsub("]", "__")
+      end
+
+      sig { params(str: String).returns(String) }
+      def escape(str)
+        str.gsub(/\W/) { |ch| ch.codepoints.map { |cp| "##{cp};" }.join }
+      end
+
+      sig { params(str: String).returns(String) }
+      def display_name(str)
+        case File.extname(str)
+        when ".rb"
+          "fa:fa-gem #{str}&nbsp;"
+        when ".css"
+          "fab:fa-css3 #{str}&nbsp;"
+        when ".png"
+          "fa:fa-image #{str}&nbsp;"
+        else
+          str
+        end
+      end
+
+      sig do
+        params(out: StringIO, node: T.untyped, path: T::Array[String]).void
+      end
+      def print_routes(out, node, path = [])
+        node.each do |key, value|
+          path2 = path + [key]
+
+          if value.is_a?(String)
+            if key == "page.rb" || key == "page.rux"
+              pathstr = path.flatten.join("/").sub(%r{\A/?}, "/")
+              out.puts "    ROUTE__#{encode(value)}[#{pathstr.inspect}]"
+            end
+          else
+            print_routes(out, value, path2)
+          end
+        end
+      end
+
+      sig do
+        params(out: StringIO, node: T.untyped, path: T::Array[String]).void
+      end
+      def print_route_edges(out, node, path = [])
+        node.each do |key, value|
+          path2 = path + [key]
+
+          if value.is_a?(String)
+            if key == "page.rb" || key == "page.rux"
```

Comment:
make this more generic or something... at least create a RE, see if it matches `/\Apage\.(?:rb|rux)\z/`